DEVELOPING.md

```diff
@@ -207,3 +213,35 @@ Please refer to the MAINTAINERS file for maintenance guidelines and contact info
 ## Contact and Support
 
 If you need help or have any questions, please open an issue on GitHub or contact us at support@neuralmagic.com.
+
+This will run automatic code styling using `black` and `isort` and test that the
+repository's code matches its standards.
+
+## Project configuration
+
+The project configuartion is powered by _[`🔗 pydantic-settings`](https://docs.pydantic.dev/latest/concepts/pydantic_settings/)_
+
+The project configuration entrypoint is represented by lazy-loaded `settigns` singleton object ( `src/config/__init__` )
+
+The project is fully configurable with environment variables. All the default values and
+
+```py
```

Comment:
I'm not sure if I'm following the purpose of having the sample code here. The code within the repo invokes immediately, right? So, we should likely have a full list of the env variables here that people would want to change alongside a link to the source like above in place of code showing them how it works

Reply:
It just reflects the idea of how this works. It might be helpful for developers who gonna work on the project. I believe that the sample of a code represents the idea better than words in most cases but I can remove it.

Reply:
This is kinda you have a code and a small description below that explains the idea on that concrete code 😄
this the _"the purpose of having the sample code here"_